protocols/raft/src/main/java/io/atomix/protocols/raft/roles/LeaderAppender.java

```diff
@@ -450,7 +450,7 @@ protected void failAttempt(RaftMemberContext member, RaftRequest request, Throwa
     // Verify that the leader has contacted a majority of the cluster within the last two election timeouts.
     // If the leader is not able to contact a majority of the cluster within two election timeouts, assume
     // that a partition occurred and transition back to the FOLLOWER state.
-    if (System.currentTimeMillis() - Math.max(getHeartbeatTime(), leaderTime) > raft.getElectionTimeout().toMillis() * 2) {
+    if (member.getFailureCount() >= 3 && System.currentTimeMillis() - Math.max(getHeartbeatTime(), leaderTime) > raft.getElectionTimeout().toMillis() * 2) {
```

Comment:
Should move this magic number to a constant.

Reply:
hmm actually `raft.getElectionTimeout().toMillis()` should also be using the existing field that already has the milliseconds computed